### PR TITLE
Custom names group actions

### LINKED ISSUE #974: History Box: Give custom names to GroupActions
In the history box, all group actions are named GroupAction. This can be kind of confusing as to what the action actually is. Ideally, the name should be representative of what the group action actually does.

This can be accomplished in a few different ways, although the simplest would probably be to add an optional parameter to the constructor that is the name of the action. If no parameter is passed in, then it can default to "Group Action".

### PR DESCRIPTION
**Pull request by jssunray34**
I added a customName variable and setter method to the GroupActions class to enable GroupActions to be assigned custom names. I also went through the codebase and gave custom names to several existing GroupActions.

Fixes #974

### CODE REVIEW

File: src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ const OscilloscopeSamplesConfig: ModuleConfig<[Oscilloscope], number> = {
     types: [Oscilloscope],
     valType: "int",
     getProps: (o) => o.getNumSamples(),
-    getAction: (s, newNumSamples) => new GroupAction(s.map(o => new OscilloscopeSamplesChangeAction(o, newNumSamples)))
+    getAction: (s, newNumSamples) => new GroupAction(s.map(o => new OscilloscopeSamplesChangeAction(o, newNumSamples)), "Oscilloscope Action")
```
Comment (TGCrystal) [inline src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx:16]: rename to `"Oscilloscope Samples Change Module"`

File: src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ const OscilloscopeInputCountConfig: ModuleConfig<[Oscilloscope], number> = {
     valType: "int",
     getProps: (o) => o.getInputPortCount().getValue(),
     getAction: (s, newCount) => (
-        new GroupAction(s.map(o => new InputPortChangeAction(o, o.getInputPortCount().getValue(),  newCount)))
+        new GroupAction(s.map(o => new InputPortChangeAction(o, o.getInputPortCount().getValue(),  newCount)), "Oscilloscope Action")
```
Comment (TGCrystal) [inline src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx:49]: rename to `"Oscilloscope Input Count Change Module"`

File: src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx [hunk 3]
```diff
@@ -66,15 +66,15 @@ const OscilloscopeDisplaySizeModuleXConfig: ModuleConfig<[Oscilloscope], number>
     valType: "float",
     getProps: (o) => o.getDisplaySize().x,
     getAction: (s, newW) => (
-        new GroupAction(s.map(o => new OscilloscopeSizeChangeAction(o, V(newW, o.getDisplaySize().y))))
+        new GroupAction(s.map(o => new OscilloscopeSizeChangeAction(o, V(newW, o.getDisplaySize().y))), "Oscilloscope Action")
```
Comment (TGCrystal) [inline src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx:69]: rename to `"Oscilloscope Width Change Module"`

File: src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx [hunk 4]
```diff
@@ -66,15 +66,15 @@ const OscilloscopeDisplaySizeModuleXConfig: ModuleConfig<[Oscilloscope], number>
...
     )
 }
 const OscilloscopeDisplaySizeModuleYConfig: ModuleConfig<[Oscilloscope], number> = {
     types: [Oscilloscope],
     valType: "float",
     getProps: (o) => o.getDisplaySize().y,
     getAction: (s, newH) => (
-        new GroupAction(s.map(o => new OscilloscopeSizeChangeAction(o, V(o.getDisplaySize().x, newH))))
+        new GroupAction(s.map(o => new OscilloscopeSizeChangeAction(o, V(o.getDisplaySize().x, newH))), "Oscilloscope Action")
```
Comment (TGCrystal) [inline src/site/pages/digital/src/containers/SelectionPopup/modules/OscilloscopeModules.tsx:77]: rename to `"Oscilloscope Height Change Module"`

**Review by TGCrystal — CHANGES_REQUESTED** [review]
Looking good! There are a few more `GroupAction`s that can be named:
- in the `src/app/core/tools/handlers` folder, they can just have names that are the same as the file name (example: the `GroupAction` in `CopyHandler.ts` could be named `Copy Handler`)
- in https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/pages/digital/src/containers/ExprToCircuitPopup/generate.ts#L112 that group action can be called `Expression Parser Action`
- https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/pages/digital/src/containers/ICDesigner/index.tsx#L169 can be named `Create IC Action`
- Modules in `src/site/pages/digital/src/containers/SelectionPopup/modules` can be named to match the file name, such as `SelectPortCountModule.tsx` can have its `GroupAction` named `Select Port Count Module`
- https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/pages/digital/src/utils/DigitalCreate.ts#L98 can be named `Smart Place Group`
- https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/pages/digital/src/utils/DigitalCreate.ts#L125 can be named `Smart Place` (don't worry about the `GroupAction`s inside of that
- https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/pages/digital/src/utils/DigitalPaste.ts#L53 can be `Recursive Transfer ICData Action`
- https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/pages/digital/src/utils/DigitalPaste.ts#L88 can be `Digital Paste`
- https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/shared/containers/ContextMenu/index.tsx#L77 can be `Cut (Context Menu)`
- https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/shared/containers/ContextMenu/index.tsx#L77 can be `Title Module`

**Comment by jssunray34** [comment]
In your first comment, is the link for `Title Module` supposed to be something else? I think you may have copy-pasted the same link from the above bullet.

**Comment by TGCrystal** [comment]
> In your first comment, is the link for `Title Module` supposed to be something else? I think you may have copy-pasted the same link from the above bullet.

You are right, here is the correct link: https://github.com/OpenCircuits/OpenCircuits/blob/master/src/site/shared/containers/SelectionPopup/modules/TitleModule.tsx#L11

**Review by TGCrystal — DISMISSED** [review]

**Review by LeonMontealegre — APPROVED** [review]
I think a lot of these names could probably be improved to read better to the user, since something like "Selection Box Tool" doesn't sound great, but we can make a separate issue for that and let have it be a "good first issue"

**Pull request merged**